src/storage/transaction/TransactionManager.cpp

```diff
@@ -192,13 +194,16 @@ void TransactionManager::onLeaderLostWrapper(const ::nebula::kvstore::Part::Call
                               opt.spaceId,
                               opt.partId,
                               opt.term);
-  auto currTermKey = std::make_pair(opt.spaceId, opt.partId);
-  auto currTermIter = currTerm_.find(currTermKey);
-  if (currTermIter == currTerm_.end()) {
-    return;
+  // clean some out-dated item in memory lock
+  for (auto cit = memLocks_.cbegin(); cit != memLocks_.cend(); ++cit) {
+    auto& [spaceId, partId, termId] = cit->first;
+    if (spaceId == opt.spaceId && partId == opt.partId && termId < opt.term) {
+      auto sptrLockCore = cit->second;
+      if (sptrLockCore->size() == 0) {
+        cit = memLocks_.erase(cit);
```

Comment:
`cit` returned by erase and `++cit ` can cause missing elements

Reply:
good catch..